### ISSUE TITLE
Show statistics for matches, not just keypoints

### ISSUE THREAD

**Issue by omgitsraven** [rhiestan/Regard3D#13]
After performing the 'compute matches' step, the minimum/median/etc number of keypoints per image is returned so we can have a sense of its success. This is handy, but it's also only half of what this step's job was; nothing indicates how many *matches* were found or how strongly they correspond, other than the peek at the top ten pairs.

I understand that the number of matches per set is usually very high and they can't all be displayed, but there needs to at least be some indication of information like "total number of matches", "avg number of matching images per image", "avg number of matching keypoints per match", "percentage of images without matches", etc...

This seems like information that must already exist somewhere if the triangulation step is possible, right? I'm working with a data set that returned 300/64318/20427/19021 for its keypoints, but consistently returns 0/393 cameras for its triangulation step, no matter what triangulation settings I choose, and it would be helpful to have a sense of what's making these images fail to correspond, when looking at the "top ten" makes the matches seem so healthy.

**Comment by rhiestan** [comment]
You can see the matches when you go to the "Show matching results" window. In the lower half, you see the matches, either putative, or the filtered matches.

**Comment by omgitsraven** [comment]
Yes but only the best ten of them! That doesn't tell me anything about what *most* of the data is like.

**Comment by rhiestan** [comment]
Yes, this is a bug in the current version of Regard3D. Should be fixed in the next version.

**Comment by omgitsraven** [comment]
Oh, good! Thank you. Still, even if I can view it all, I still think it would be handy for the program to summarize these statistics about it, so I don't have to count things manually for each set of matches.

**Comment by rhiestan** [comment]
To be honest, I wouldn't know how to summarize the filtered matches in a small text. Do you have a suggestion?

**Comment by omgitsraven** [comment]
I made my suggestions in my original post...

- Total number of matching image pairs
- Average number of matching keypoints in each image pair
- Average number of other images that each image matches with
- Percentage of images that don't match with anything (or that have less than 1% of the avg amount of matches, or something)

I'm new to this, so maybe there's things I'm not thinking of, but that seems like a decent overview of how healthy the data set is? A decent way to compare at a glance how well different settings worked... (It would be even better if you could give min/max/avg for the average ones too...)
